packages/trader/src/Modules/Contract/Components/LastDigitPrediction/__tests__/digit-spot.spec.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import DigitSpot from '../digit-spot';
+
+const mocked_props: React.ComponentProps<typeof DigitSpot> = {
+    current_spot: '12345',
+};
+
+describe('<DigitSpot />', () => {
+    it('should render current_spot value without the last digit ', () => {
+        render(<DigitSpot {...mocked_props} />);
+        const sliced_spot_value = screen.getByText('1234');
```

Comment:
How do you check, that last digit (`5`) is not in the current spot? Because in next test case you render `<DigitSpot />` with the same `mocked_props` and you're able to find `5` (line 18). Just asking

Reply:
` <Text
            size='xs'
            align='center'
            weight='bold'
            className='digits__digit-spot-value'
            data-testid='dt_digits_digit_spot_value'
        >
            {current_spot ? current_spot.slice(0, -1) : ''}
        </Text>
        <span
            className={classNames('digits__digit-spot-last', {
                'digits__digit-spot-last--selected-win': is_selected_winning,
                'digits__digit-spot-last--win': is_won,
                'digits__digit-spot-last--loss': is_lost,
            })}
            data-testid='dt_digits_digit_spot_last'
        >
            {current_spot ? current_spot.slice(-1) : ''}
        </span>`
        
 
 
 The first current_spot.slice returns current_spot price minus the last digit for example 12345 to <Text>1234</Text>
 
 The second current_spot.slice returns last digit of current_spot so 12345 returns just <span>5</span>
 
 To answer your question you just do `screen.queryByText('12345').not.toBeInTheDocument()` i guess lol

Reply:
Yep, I now about slice(-1), but if it doesn't work for some reason and the last digit won't be sliced, your test will pass anyway, won't it?

Reply:
If slice doesn't work then my test wont pass because i'm doing '1234' instead of /1234/.

'1234' looks for exactly 1234.
/1234/ looks for anything that has 1234 even if the number is 123456789 it'll pass.